Mathlib/MeasureTheory/Group/ModularCharacter.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2025 Noam Atar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Noam Atar
+-/
+import Mathlib.MeasureTheory.Function.LocallyIntegrable
+import Mathlib.MeasureTheory.Group.Integral
+import Mathlib.MeasureTheory.Group.Measure
+import Mathlib.Topology.Metrizable.Urysohn
+import Mathlib.MeasureTheory.Measure.Haar.Unique
+import Mathlib.MeasureTheory.Constructions.BorelSpace.Basic
+
+/-!
+# Modular character of a locally compact group
+
+On a locally compact group, there is a natural homomorphism `G → ℝ≥0*`, which for `g : G` gives the
+value `μ (· * g⁻¹) / μ`, where `μ` is an (inner regular) Haar measure. This file defines this
+homomorphism, called the modular character, and shows that it is independent of the chosen Haar
+measure.
+
+TODO: Show that the character is continuous.
+
+## Main Declarations
+
+* `modularCharacterFun`: Define the modular character function. If `μ` is a left Haar measure on `G`
+  and `g : G`, the measure `A ↦ μ (A g⁻¹)` is also a left Haar measure, so by uniqueness is of the
+  form `Δ(g) μ`, for `Δ(g) ∈ ℝ≥0`. This `Δ` is the modular character. The result that this does not
+  depend on the measure chosen is `modularCharacterFun_eq_haarScalarFactor`.
+* `modularCharacter`: The homomorphism G →* ℝ≥0 whose toFun is `modularCharacterFun`.
+-/
+
+open MeasureTheory
+open scoped NNReal
+
+namespace MeasureTheory
+
+namespace Measure
+
+variable {G : Type*} [TopologicalSpace G] [Group G] [IsTopologicalGroup G] [LocallyCompactSpace G]
+
+/-- The modular character as a map is `g ↦ μ (· * g⁻¹) / μ`, where `μ` is a left Haar measure.
+
+  See also `modularCharacter` that defines the map as a homomorphism. -/
+@[to_additive "The additive modular character as a map is `g ↦ μ (· + -g) / μ`, where `μ` is an
+  left additive Haar measure.
+
+  See also `modularCharacter` that defines the map as a multiplicative homomorphism.."]
```

Comment:
```suggestion
  See also `addModularCharacter` that defines the map as an additive homomorphism.."]
```

Reply:
I wrote this because I couldn't define `addModularCharacter` without using something like `Multiplicative G \to ℝ≥0` (and in this case `to_additive` wouldn't recognize it), so I just left only `modularCharacter`.

Reply:
Ah, yes. Then just remove the sentence.